api/test_api.py

```diff
@@ -96,3 +97,52 @@ def test_me_endpoint(mock_get_current_user):
     assert response.status_code == 200
     assert ('_id' and 'username' and 'hashed_password' and
             'active') in response.json()
+
+
+@pytest.fixture()
+def mock_init_sub_id(mocker):
+    async_mock = AsyncMock()
+    mocker.patch('api.pubsub.PubSub._init_sub_id',
+                 side_effect=async_mock)
+    return async_mock
+
+
+@pytest.fixture()
+def mock_subscribe(mocker):
+    async_mock = AsyncMock()
+    mocker.patch('api.pubsub.PubSub.subscribe',
+                 side_effect=async_mock)
+    return async_mock
+
+
+def test_subscribe_endpoint(mock_get_current_user, mock_init_sub_id,
+                            mock_subscribe):
+    """
+    Test Case : Test KernelCI API /subscribe endpoint
+    Expected Result :
+        HTTP Response Code 200 OK
+        JSON with 'id' and 'channel' keys
+    """
+    user = User(username='bob',
+                hashed_password='$2b$12$CpJZx5ooxM11bCFXT76/z.o6HWs2sPJy4iP8.'
+                                'xCZGmM8jWXUXJZ4K',
+                active=True)
+    mock_get_current_user.return_value = user
+
+    subscribe = Subscription(id=1, channel='abc')
+    mock_subscribe.return_value = subscribe
+
+    with TestClient(app) as client:
+        # Used 'TestClient' with a 'with' to trigger startup event on app
```

Comment:
It may be more accurate to use an actual name of the language feature i.e. `context manager` instead of `'with'`
e.g.
> Use context manager to trigger a startup event on the app object

Reply:
Modified the comment.